src/term.c

```diff
@@ -307,14 +313,22 @@ static int cmd_dump(PROGRAMMER *pgm, AVRPART *p, int argc, char *argv[]) {
         read_mem[i].addr = 0;
       read_mem[i].len = maxsize - read_mem[i].addr;
     } else if (argc == 4) {
-      unsigned long ul = strtoul(argv[3], &end_ptr, 0);
+      int len = strtol(argv[3], &end_ptr, 0);
       if (*end_ptr || (end_ptr == argv[3])) {
         pmsg_error("(dump) cannot parse length %s\n", argv[3]);
         return -1;
       }
-      if (ul == 0 || ul > INT_MAX) // Not positive if used as int, make it 1
-        ul = 1;
-      read_mem[i].len = (int) ul;
+      // Turn negative len value (no. bytes from top of memory) into an actual length number
+      if (len < 0)
+        len = maxsize + len + 1 - read_mem[i].addr;
+
+      if (len == 0)
+        return 0;
+      else if (len < 0) {
```

Comment:
This is not wrong, but I normally refrain from elsing a return

Reply:
Sorry, that's a mistake on my side. I'll have it fixed